### PR TITLE
PI-21796: Basic Auth POC in SDK

### PR DESCRIPTION
**Pull request by rajbhupendra588**
#### Dependencies
[//]: # "optional, remove if none"
List additional Jira tickets or Github PRs

#### Description
Description of the PR. Make language appropriate for other engineers not involved.
-Basic Auth POC in SDK

### CODE REVIEW

File: src/main/java/com/appdirect/sdk/web/oauth/BasicOauthSecurityConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
+
+import com.appdirect.sdk.appmarket.DeveloperSpecificAppmarketCredentialsSupplier;
+import com.appdirect.sdk.web.oauth.model.OpenIdCustomUrlPattern;
+
+@Configuration
+@EnableWebSecurity
+@Slf4j
+@Order(1)
```
Comment (ketanmulay-appdirect) [inline src/main/java/com/appdirect/sdk/web/oauth/BasicOauthSecurityConfiguration.java:52]: Basic auth should be of least priority. Please add @Order(101) to this class and @Order(100) on com.appdirect.sdk.web.oauth.SecurityConfiguration

File: src/main/java/com/appdirect/sdk/web/oauth/BasicOauthSecurityConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+		return baiscOauthService().getBasicFilter();
+	}
+
+	@Bean
+	public ConsumerDetailsService consumerDetailsService() {
+		return new DeveloperSpecificAppmarketCredentialsConsumerDetailsService(credentialsSupplier);
+	}
+
+	@Bean
+	public OAuthProviderTokenServices oauthProviderTokenServices() {
```
Comment (ketanmulay-appdirect) [inline src/main/java/com/appdirect/sdk/web/oauth/BasicOauthSecurityConfiguration.java:81]: Why do we need token service for basic auth?

File: src/main/java/com/appdirect/sdk/web/oauth/BaiscOauthService.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package com.appdirect.sdk.web.oauth;
+
+import javax.servlet.Filter;
+
+public interface BaiscOauthService {
```
Comment (ketanmulay-appdirect) [inline src/main/java/com/appdirect/sdk/web/oauth/BaiscOauthService.java:5]: Please change name of the interface to BasicAuthService. We dont need oAuth here.

File: src/main/java/com/appdirect/sdk/web/oauth/BaiscOauthServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package com.appdirect.sdk.web.oauth;
+
+import javax.servlet.Filter;
+
+public class BaiscOauthServiceImpl implements BaiscOauthService {
```
Comment (ketanmulay-appdirect) [inline src/main/java/com/appdirect/sdk/web/oauth/BaiscOauthServiceImpl.java:5]: Please correct class name here as well.

File: src/main/java/com/appdirect/sdk/web/oauth/BaiscOauthSupplier.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import javax.servlet.Filter;
+
+/**
+ * Implementations of this interface provide a way for the service-integration-sdk
+ * to retrieve the Basic filter. Each SDK client application must contain a bean
+ * of this type in its application context in order for the communication with AppMarket to work.
+ */
+
+@FunctionalInterface
+public interface BaiscOauthSupplier {
```
Comment (ketanmulay-appdirect) [inline src/main/java/com/appdirect/sdk/web/oauth/BaiscOauthSupplier.java:12]: Please correct name here. Do you have corresponding connector implementation for this?

File: src/main/java/com/appdirect/sdk/web/oauth/BaiscOauthSupplierImpl.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
+package com.appdirect.sdk.web.oauth;
+
+import javax.servlet.Filter;
+
+public class BaiscOauthSupplierImpl implements  BaiscOauthSupplier{
```
Comment (ketanmulay-appdirect) [inline src/main/java/com/appdirect/sdk/web/oauth/BaiscOauthSupplierImpl.java:5]: Please change name.

File: src/main/java/com/appdirect/sdk/web/oauth/BasicOauthSecurityConfiguration.java [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+		return new CoreOAuthProviderSupport();
+	}
+
+	@Bean
+	public OAuthKeyExtractor oauthKeyExtractor() {
+		return new OAuthKeyExtractor(oauthProviderSupport());
+	}
+
+	@Bean
+	public ProtectedResourceProcessingFilter oAuthSignatureCheckingFilter() {
```
Comment (ketanmulay-appdirect) [inline src/main/java/com/appdirect/sdk/web/oauth/BasicOauthSecurityConfiguration.java:101]: do we need to check oAuthSignature?

File: src/main/java/com/appdirect/sdk/web/oauth/BaiscOauthSupplier.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+	/**
+	 * Returns the Basic Filter
+	 *
+	 * @return the Filter to authorize incoming requests
+	 */
+	Filter getBasicFilter();
```
Comment (ketanmulay-appdirect) [inline src/main/java/com/appdirect/sdk/web/oauth/BaiscOauthSupplier.java:18]: please rename to getBasicAuthFilter

File: src/main/java/com/appdirect/sdk/web/oauth/BasicOauthSecurityConfiguration.java [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+		return new OpenIdCustomUrlPattern();
+	}
+
+	@Bean
+	public BaiscOauthService baiscOauthService() {
+		return new BaiscOauthServiceImpl(baiscOauthSupplier);
+	}
+
+	@Bean
+	public Filter basicSignatureCheckingFilter() {
```
Comment (ketanmulay-appdirect) [inline src/main/java/com/appdirect/sdk/web/oauth/BasicOauthSecurityConfiguration.java:71]: Please rename to basicAuthenticationFilter

**Review by ketanmulay-appdirect — CHANGES_REQUESTED** [review]

**Review by abidhkhan — APPROVED** [review]

**Comment by rajbhupendra588** [comment]
Dev Testing:- 

**Basic Auth :-** 
**With Valid username & password:-** 

image


**With InValid username & password:-** 
image



**Auth1 :-** [Not breaking] 
image

File: src/main/java/com/appdirect/sdk/appmarket/events/AppmarketEventService.java [hunk 10]
```diff
@@ -61,6 +66,56 @@ APIResult processEvent(String eventUrl, EventHandlingContext eventContext) {
...
+				byte[] credDecoded = Base64.getDecoder().decode(base64Credentials);
+				credentials = new String(credDecoded, StandardCharsets.UTF_8);
+				log.info("credentials={}",credentials);
+				// credentials = username:password
+				values = credentials.split(":", 2);
+				usr = values[0];
+				pass = values[1];
+
+				log.info("Username ->{}",usr);
+				log.info("Password ->{}",pass);
```
Comment (ketanmulay-appdirect) [inline src/main/java/com/appdirect/sdk/appmarket/events/AppmarketEventService.java:91]: Please remove password from logs

**Review by ketanmulay-appdirect — CHANGES_REQUESTED** [review]

**Review by reshmakadam-appdirect — DISMISSED** [review]

**Review by nsadaphal — DISMISSED** [review]
LGTM

File: src/main/java/com/appdirect/sdk/appmarket/DeveloperSpecificAppmarketBasicAuthCredentialsSupplier.java [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+
+package com.appdirect.sdk.appmarket;
+
+/**
+ * Implementations of this interface provide a way for the service-integration-sdk
+ * to retrieve the developer credentials. Each SDK client application must contain a bean
+ * of this type in its application context in order for the communication with AppMarket to work.
+ */
+@FunctionalInterface
+public interface DeveloperSpecificAppmarketBasicAuthCredentialsSupplier {
```
Comment (ketanmulay-appdirect) [inline src/main/java/com/appdirect/sdk/appmarket/DeveloperSpecificAppmarketBasicAuthCredentialsSupplier.java:22]: Can you please rename it to BasicAuthCredentialsSupplier? For new classes inserted we are not following old naming convention.

File: src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificAppmarketBasicAuthCredentialsConsumerDetailsService.java [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+
+package com.appdirect.sdk.web.oauth;
+
+import org.springframework.security.oauth.provider.ConsumerDetails;
+import org.springframework.security.oauth.provider.ConsumerDetailsService;
+
+import com.appdirect.sdk.appmarket.Credentials;
+import com.appdirect.sdk.appmarket.DeveloperSpecificAppmarketBasicAuthCredentialsSupplier;
+
+class DeveloperSpecificAppmarketBasicAuthCredentialsConsumerDetailsService implements ConsumerDetailsService {
```
Comment (ketanmulay-appdirect) [inline src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificAppmarketBasicAuthCredentialsConsumerDetailsService.java:22]: Please rename to BasicAuthCredentialsConsumerDetailsService

**Review by ketanmulay-appdirect — CHANGES_REQUESTED** [review]

**Comment by rajbhupendra588** [comment]
Dev Testing :- 



**Basic Auth with Valid credential :-** 

image





**Basic Auth with InValid credential:-** 
 
image





**OAuth1 with Valid credential :-**  

image

**Review by raushan-appdirect — APPROVED** [review]
LGTM

**Review by nsadaphal — APPROVED** [review]

**Review by ketanmulay-appdirect — APPROVED** [review]
LGTM

**Pull request merged**